cmd/servicelog/common.go

```
@@ -73,59 +74,73 @@ func applyFilters(ocmClient *sdk.Connection, filters []string) ([]*v1.Cluster, e
 func sendRequest(request *sdk.Request) *sdk.Response {
 	response, err := request.Send()
 	if err != nil {
-		log.Fatalf("Can't send request: %v", err)
+		responseMessage = fmt.Sprintf("Cannot send request: %q", err)
+		return nil
 	}
 	return response
 }
 
-func check(response *sdk.Response, clusterMessage servicelog.Message) {
+func check(response *sdk.Response, clusterMessage servicelog.Message) bool {
 	body := response.Bytes()
 	if response.Status() < 400 {
-		validateGoodResponse(body, clusterMessage)
-		log.Infof("Message has been successfully sent to %s\n", clusterMessage.ClusterUUID)
+		goodReply := validateGoodResponse(body, clusterMessage)
+		return goodReply != nil
 	} else {
-		badReply := validateBadResponse(body)
-		log.Fatalf("Failed to post message because of %q", badReply.Reason)
+		validateBadResponse(body)
+		return false
 	}
 }
 
-func validateGoodResponse(body []byte, clusterMessage servicelog.Message) servicelog.GoodReply {
-	var goodReply servicelog.GoodReply
+func validateGoodResponse(body []byte, clusterMessage servicelog.Message) *servicelog.GoodReply {
```

You can also return a error in this func if the validation failed.